ignite/metrics/ssim.py

```diff
@@ -161,6 +161,9 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
         y_pred = F.pad(y_pred, [self.pad_w, self.pad_w, self.pad_h, self.pad_h], mode="reflect")
         y = F.pad(y, [self.pad_w, self.pad_w, self.pad_h, self.pad_h], mode="reflect")
 
+        if y_pred.dtype != torch.get_default_dtype():
+            self._kernel = self._kernel.to(dtype=y_pred.dtype)
```

Comment:
Should not we compare to `self._kernel.dtype` here ?
```suggestion
        if y_pred.dtype != self._kernel.dtype:
            self._kernel = self._kernel.to(dtype=y_pred.dtype)
```

Reply:
we can, but I think they always are going to be equal

```python
self._kernel.dtype == torch.get_default_dtype()
>>> True
```

Reply:
but that would make the change more clear to others, so maybe it is a good idea

Reply:
If user explicitly define y and y_pred with torch.float64 dtype and `torch.get_default_dtype()` will be still float32, so the error will remain.